src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -0,0 +1,1178 @@
+  navigation.push_back( {{ "title",  "Collections" }, { "href", parentLink( url, 1 ) }} ) ;
+  write( data, context, {{ "pageTitle", title }, { "navigation", navigation }} );
+}
+
+json Wfs3DescribeCollectionHandler::schema( const QgsServerApiContext &context ) const
+{
+  json data;
+  Q_ASSERT( context.project() );
+
+  const auto layers { QgsServerApiUtils::publishedWfsLayers<QgsVectorLayer>( context.project() ) };
+  // Construct the context with collection id
+  for ( const auto &mapLayer : layers )
+  {
+    const auto shortName { mapLayer->shortName().isEmpty() ? mapLayer->name() : mapLayer->shortName() };
+    const auto title { mapLayer->title().isEmpty() ? mapLayer->name().toStdString() : mapLayer->title().toStdString() };
+    const auto path { ( context.apiRootPath() + QStringLiteral( "collections/%1" ).arg( shortName ) ).toStdString() };
+
+    data[ path ] =
+    {
+      {
+        "get", {
+          { "tags", jsonTags() },
+          { "summary", "Describe the '" + title + "' feature collection"},
+          { "operationId", operationId() + '_' + shortName.toStdString() },
+          {
+            "responses", {
+              {
+                "200", {
+                  { "description", "Metadata about the collection '" + title + "' shared by this API." },
+                  {
+                    "content", {
+                      {
+                        "application/json", {
+                          {
+                            "schema",  {
+                              { "$ref", "#/components/schemas/collectionInfo" }
+                            }
+                          }
+                        }
+                      },
+                      {
+                        "text/html", {
+                          {
+                            "schema",  {
+                              { "type", "string" }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              },
+              defaultResponse()
+            }
+          }
+        }
+      }
+    };
+  }
+  return data;
+}
+
+Wfs3CollectionsItemsHandler::Wfs3CollectionsItemsHandler()
+{
+}
+
+QList<QgsServerQueryStringParameter> Wfs3CollectionsItemsHandler::parameters( const QgsServerApiContext &context ) const
+{
+  QList<QgsServerQueryStringParameter> params;
+
+  // Limit
+  QgsServerQueryStringParameter limit { QStringLiteral( "limit" ), false,
+                                        QgsServerQueryStringParameter::Type::Integer,
+                                        QStringLiteral( "Number of features to retrieve [0-10000]" ),
+                                        10 };
+  limit.setCustomValidator( [ ]( const QgsServerApiContext &, QVariant & value ) -> bool
+  {
+    return value >= 0 && value <= 10000;   // TODO: make this configurable!
```

Comment:
Can you define this limit as a const integer at the beginning of the file please?

Reply:
Yes, this is a TODO. I'd like to make this an env configurable variable, but for now I'll move the integer to a const as you suggested.